### PR TITLE
chore!: Remove unused code/features: Global tables, CascadeDeleteFilters, ExtraFields, AlwaysDelete

### PR DESCRIPTION
**Pull request by disq**
Completes https://github.com/cloudquery/cq-provider-sdk/issues/308

#273 #309 #310 #311

### CODE REVIEW

**Review by yevgenypats — APPROVED** [review]
Love code deletion!

File: cqproto/internal/plugin.proto [hunk 1]
```diff
@@ -24,8 +24,6 @@ message ConfigureProvider {
     ConnectionDetails connection = 2;
     // Holds information such as credentials, regions, accounts, etc'
     bytes config = 3;
-    // Msgpack encoded: Allows to inject & override fields into resource tables, use this carefully to override fields. Used in tests.
-    bytes extraFields = 5;
```
Comment (roneli) [inline cqproto/internal/plugin.proto:28]: not 100% sure, but wouldn't this break prior versions? wouldn't it be better to do [deprecated = true] instead, and when decoding / encoding with new SDK ignore it?

File: provider/schema/table.go [hunk 2]
```diff
@@ -43,19 +43,13 @@ type Table struct {
...
 
 	// IgnoreInTests is used to exclude a table from integration tests.
 	// By default, integration tests fetch all resources from cloudquery's test account, and verifY all tables
 	// have at least one row.
 	// When IgnoreInTests is true, integration tests won't fetch from this table.
 	// Used when it is hard to create a reproducible environment with a row in this table.
 	IgnoreInTests bool
-	// Global tables are usually the same regardless of the provider fetch configuration. Global table data gets fetched
-	// and doesn't produce PK conflict errors instead data is replaced
-	Global bool
```
Comment (roneli) [inline provider/schema/table.go:58]: nit: would require a PR to AWS provider, same with AlwaysDelete

**Review by roneli — APPROVED** [review]
LGTM, had two comments. The second is stating the obvious ^^, so just auto-resolve after reading it.

File: provider/schema/table.go (hunk 2 above)
Comment (disq) [reply provider/schema/table.go:58]: https://github.com/cloudquery/cq-provider-aws/pull/1124

File: cqproto/internal/plugin.proto (hunk 1 above)
Comment (disq) [reply cqproto/internal/plugin.proto:28]: it would just show up as default empty value which would be a nil slice and map to a nil map.

**Pull request merged**